### PR TITLE
feat: overhaul virtual circuits

### PR DESCRIPTION
**Pull request by umut-sahin**

### CODE REVIEW

File: concrete/numpy/compilation/circuit.py [hunk 1]
```diff
@@ -65,6 +69,36 @@ def __init__(self, graph: Graph, mlir: str, configuration: Optional[Configuratio
...
+        Returns:
+            Any:
+                result of the simulation
+        """
+
+        return self.graph(*args, p_error=self.configuration.p_error)
+
+    def to_fhe(self):
+        """
+        Compile the circuit if it is virtual, do nothing otherwise.
```
Comment (rudy-6-4) [inline concrete/numpy/compilation/circuit.py:89]: The doc is misleading.
This change/reconfigure the circuit
Convert/Reconfigure a virtual circuit to a compiled/real one or do nothing if already the case.

File: concrete/numpy/compilation/circuit.py [hunk 2]
```diff
@@ -65,6 +69,36 @@ def __init__(self, graph: Graph, mlir: str, configuration: Optional[Configuratio
...
+                inputs to the circuit
+
+        Returns:
+            Any:
+                result of the simulation
+        """
+
+        return self.graph(*args, p_error=self.configuration.p_error)
+
+    def to_fhe(self):
```
Comment (rudy-6-4) [inline concrete/numpy/compilation/circuit.py:87]: the name could be better
reconfigure_XX / convert_XX / upgrade_XX

**Review by rudy-6-4 — APPROVED** [review]

File: concrete/numpy/compilation/circuit.py (hunk 1 above)
Comment (umut-sahin) [reply concrete/numpy/compilation/circuit.py]: Updated.

File: concrete/numpy/compilation/circuit.py (hunk 2 above)
Comment (umut-sahin) [reply concrete/numpy/compilation/circuit.py]: Renamed to `enable_fhe`.

**Review by rudy-6-4 — APPROVED** [review]
You were right, I love the doc.

**Pull request merged**